monai/metrics/meandice.py

```diff
@@ -77,22 +118,21 @@ def __init__(
             include_background=self.include_background,
             reduction=MetricReduction.NONE,
             get_not_nans=False,
-            softmax=False,
+            apply_argmax=False,
             ignore_empty=self.ignore_empty,
             num_classes=self.num_classes,
         )
 
     def _compute_tensor(self, y_pred: torch.Tensor, y: torch.Tensor) -> torch.Tensor:  # type: ignore[override]
         """
+        Compute the dice value using ``DiceHelper``.
+
         Args:
-            y_pred: input data to compute, typical segmentation model output.
-                It must be one-hot format and first dim is batch, example shape: [16, 3, 32, 32]. The values
-                should be binarized.
-            y: ground truth to compute mean Dice metric. `y` can be single-channel class indices or
-                in the one-hot format.
```

Comment:
These restrictions are not really necessary, such as binarized, or one-hot, etc, right?

Reply:
This is coded into the way the `DiceHelper` class computes the metrics on a per-item and per-channel basis so it is expected. [Here](https://github.com/Project-MONAI/MONAI/blob/90de55b1340ffc8eccf960d111229535bcb2f81d/monai/metrics/meandice.py#L274) and the next line down there's assumptions about the shape of the incoming tensors.